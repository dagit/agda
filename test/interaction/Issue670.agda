-- Andreas, 2012-06-30, case reported by Noam Zeilbereger
module Issue670 where

infix 4 _≡_

data _≡_ {A : Set} (x : A) : A → Set where
  refl : x ≡ x

data Nat : Set where
  zero : Nat
  suc  : Nat → Nat

{-# BUILTIN NATURAL Nat  #-}
{-# BUILTIN SUC     suc  #-}
{-# BUILTIN ZERO    zero #-}

record T : Set where
    field
      x : Nat
      .eq : x ≡ x

foo : T
foo = record { x = 0 ; eq = refl }

bar : T
bar = record { x = 0 ; eq = {!refl!} }
-- give should succeed
