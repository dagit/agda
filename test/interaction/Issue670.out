Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue670 (Issue670.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue670.\n" t)
(agda2-status-action "Checked")
(agda2-info-action "*All Goals*" "?0 : 0 â‰¡ 0\n" nil)
((last . 1) . (agda2-goals-action '(0)))
Agda2> (agda2-give-action 0 "refl")
(agda2-status-action "Checked")
(agda2-info-action "*All Goals*" "" nil)
((last . 1) . (agda2-goals-action '()))
Agda2> 